src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts

```diff
@@ -61,6 +61,22 @@ class TerminalSuggestContribution extends DisposableStore implements ITerminalCo
 			this._pwshAddon?.dispose();
 		}));
 		this._terminalSuggestWidgetVisibleContextKey = TerminalContextKeys.suggestWidgetVisible.bindTo(this._contextKeyService);
+		this.add(this._configurationService.onDidChangeConfiguration(e => {
+			if (e.affectsConfiguration(TerminalSuggestSettingId.EnableExtensionCompletions) || e.affectsConfiguration(TerminalSuggestSettingId.Enabled)) {
+				const extensionCompletionsEnabled = this._configurationService.getValue<ITerminalSuggestConfiguration>(terminalSuggestConfigSection).enableExtensionCompletions;
+				const completionsEnabled = this._configurationService.getValue<ITerminalSuggestConfiguration>(terminalSuggestConfigSection).enabled;
+				if (!extensionCompletionsEnabled || !completionsEnabled) {
+					this._addon.clear();
+					this._pwshAddon.clear();
+				}
```

Comment:
Isn't the right condition now this?

```ts
				if (!extensionCompletionsEnabled || !completionsEnabled) {
					this._addon.clear();
				}
				if (!completionsEnabled) {
					this._pwshAddon.clear();
				}
```

Reply:
we can clear both bc below will create if neeeded

Reply:
oh nvm, I thought we checked those settings in `_loadAddons`, but we don't.